src/Covenant/Internal/Rename.hs

```diff
@@ -211,8 +316,10 @@ renameDataDecl (DataDeclaration tn cnt ctors strat) = RenameM $ do
     renameCtor :: Constructor AbstractTy -> RenameM (Constructor Renamed)
     renameCtor (Constructor cn args) = Constructor cn <$> traverse renameValT args
 
+-- REVIEW: I am not sure if we really want the scope arg to runRenameM to be `mempty`.
+--         If something breaks w/ BB forms or datatypes, look here.
```

Comment:
I'm fairly sure renaming datatypes should also be contextual. Happy to punt that to a future problem though.